packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx

```diff
@@ -60,6 +67,32 @@ export const MediaGallery = (props: MediaGalleryProps): JSX.Element => {
     };
   }, [cameraSwitcherCallback, cameraSwitcherCameras]);
 
+  useMemo(() => {
+    const onPersonJoined = (e) => {
+      setAnnouncerString(locale.participantJoinedNoticeString);
+      setInterval(() => {
+        setAnnouncerString('');
+      }, 5000);
+    };
+    adapter.on('participantsJoined', onPersonJoined);
+    return () => {
+      adapter.off('participantsJoined', onPersonJoined);
+    };
+  }, []);
+
+  useMemo(() => {
+    const onPersonLeft = (e) => {
+      setAnnouncerString(locale.participantLeftNoticeString);
+      setInterval(() => {
+        setAnnouncerString('');
```

Comment:
what if a participant joins and another one leaves?

Reply:
The announcer should just read the first string. this is something we should test on the platforms to understand rapid string changes in that component.

Reply:
I should rephrase that, what if a user leaves, then 4999ms later a participant joins (i.e. it sets setAnnouncerString('new participant') then immediately calls setAnnouncerString(''))